src/Integration.Vsix/Analysis/IssueConsumerFactory_IssueHandler.cs

```diff
@@ -42,39 +40,39 @@ internal class IssueHandler
             private readonly ITextDocument textDocument;
             private readonly string projectName;
             private readonly Guid projectGuid;
-            private readonly IIssuesFilter issuesFilter;
+            private readonly IClientSuppressionSynchronizer clientSuppressionSynchronizer;
             private readonly SnapshotChangedHandler onSnapshotChanged;
 
             private readonly TranslateSpans translateSpans;
 
             public IssueHandler(ITextDocument textDocument,
                 string projectName,
                 Guid projectGuid,
-                IIssuesFilter issuesFilter,
+                IClientSuppressionSynchronizer clientSuppressionSynchronizer,
                 SnapshotChangedHandler onSnapshotChanged)
-                : this (textDocument, projectName, projectGuid, issuesFilter, onSnapshotChanged, DoTranslateSpans)
+                : this(textDocument, projectName, projectGuid, clientSuppressionSynchronizer, onSnapshotChanged, DoTranslateSpans)
             {
             }
 
             internal /* for testing */ IssueHandler(ITextDocument textDocument,
                 string projectName,
                 Guid projectGuid,
-                IIssuesFilter issuesFilter,
+                IClientSuppressionSynchronizer clientSuppressionSynchronizer,
                 SnapshotChangedHandler onSnapshotChanged,
                 TranslateSpans translateSpans)
             {
                 this.textDocument = textDocument;
                 this.projectName = projectName;
                 this.projectGuid = projectGuid;
-                this.issuesFilter = issuesFilter;
+                this.clientSuppressionSynchronizer = clientSuppressionSynchronizer;
                 this.onSnapshotChanged = onSnapshotChanged;
 
                 this.translateSpans = translateSpans;
             }
 
             internal /* for testing */ void HandleNewIssues(IEnumerable<IAnalysisIssueVisualization> issues)
             {
-                MarkSuppressedIssues(issues);
+                clientSuppressionSynchronizer.SynchronizeSuppressedIssues();
```

Comment:
The one thing I don't like about this is that now it will synchronize all issues, instead of just the ones that we receive here. Should we consider adding an extra method for SynchronizeSuppressedIssues that has a parameter?

Reply:
> The one thing I don't like about this is that now it will synchronize all issues, instead of just the ones that we receive here

That's definitely a problem for two reasons:
1) the client synchronizer raises an event which will trigger refreshes of bits of the UI, and
2) some analyzers stream issues i.e. report one issue at a time to the `IssueConsumer`. The combination of (1) and (2) would mean for a file with 100 issue the UI refresh would be triggered and additional 100 times.


Adding an overload doesn't feel quite right either. It wouldn't need to raise the `LocalSuppressionChanged` event, so the client synchronizer would then have two methods: one that synch-ed the issues from the store and raised and event, and one that sync-ed a supplied list of issues and didn't raise an event.

There is something wrong with the existing split of responsibilities, but changing the issue handler to use the synchronizer doesn't feel quite right either. Perhaps we should discuss this on Monday.